### DISCUSSION TITLE
Will an EntityRuler label annotations affect the way a NER pipe checks if it's entities are right?

### DISCUSSION THREAD

**Discussion by Vfgandara** [explosion/spaCy#9803] [Help: Model Advice] [answered]
Hi!

As stated in the title, I'd like to better understand if having an EntityRuler will impact the performance of a NER pipe later in the pipeline.
To give more context: I'm working on a project where _technically_ we have a finite set of words that we're looking forward to label (just one label too, btw). For some other reasons that I can't disclose due to a NDA, we're still going to use a transformer with a NER pipe to do that task. But we're thinking about still adding the EntityRuler before the NER on the pipeline, just as a redundancy and to get a more reliable approach. The question is the following: in that case wouldn't the NER be unable to set entities for tokens that were already labelled by the EntityRuler? And if that's so, then wouldn't it happen that the NER model would get the feedback that it labelled the token wrong (which wouldn't be the case)? Just to be clear, we still want the NER to be able to learn properly, even the tokens that would otherwise be caught by the EntityRuler.

As always, thanks in advance for the help! You all do a great work at helping everyone at how to use this wonderful library!

**Comment by polm** [comment, marked as answer]
You can use an NER component and an EntityLinker in the same pipeline and the EntityLinker will not affect accuracy during training normally. Components can't see other components' annotations unless you explicitly make them [annotating components](https://spacy.io/usage/training#annotating-components). 

Given your description of the task, I would put the EntityLinker *after* the NER component and set it to overwrite. That said, given your description of your task, I'm still confused why you need an NER component at all - are there some times you don't want to label words?
